File: MBStore/Models/CartItemProduct.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace MBStore.Models
{
    public class CartItemProduct
    {
        public Guid CartItemId { get; set; }
        public CartItem CartItem { get; set; }

        public Guid ProductId { get; set; }
        public Product Product { get; set; }
    }
}

File: MBStore/ClientApp/src/app/shared/models/CartItemProduct.ts
import { CartItem } from './CartItem';
import { Product } from './Product';

export class CartItemProduct {
  public cartItemId: string;
  public cartItem: CartItem;

  public productId: string;
  public product: Product;

  constructor(cartItem: CartItem, product: Product) {

  }

}
